File: frontend/src/models/StationOpeningHours.ts
export interface StationOpeningHours {
    id: number;
    station_id: number;
    day_of_week: "mon" | "tue" | "wed" | "thu" | "fri" | "sat" | "sun" | "holiday";
    open_time: string;   // np. "06:00"
    close_time: string;  // np. "21:00"
    created_at: string;
    updated_at: string;
  }
  

File: backend/app/models/station_opening_hours.py
from sqlalchemy import Column, Integer, String, Time, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from app.core.db import Base
from datetime import datetime

class StationOpeningHours(Base):
    __tablename__ = "station_opening_hours"

    id = Column(Integer, primary_key=True, index=True)
    station_id = Column(Integer, ForeignKey("stations.id"), nullable=False)
    day_of_week = Column(String, nullable=False)  # "mon", "tue", ..., "sun"
    open_time = Column(Time, nullable=False)
    close_time = Column(Time, nullable=False)

    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    station = relationship("Station", back_populates="opening_hours")
